### ISSUE TITLE
Consider assignments when calculating availabilty

### ISSUE THREAD

**Issue by kgf** [ReadyResponder/ReadyResponder#208]
Currently, availability for an event is calculated simply by looking for an Availability record for the same time span.
https://github.com/ReadyResponder/ReadyResponder/blob/development/app/models/event.rb#L43
However, if someone gets an assignment, they are no longer available, they have an assignment. This needs to be considered. 
This is on hold, because more consideration is needed around overlapping availabilities, which one wins. ?

**Comment by kgf** [comment]
For now, if there are two availabilities, "Unavailable" should win. We may re-visit this in the future, but I would like to get this issue advancing.
Plus, it's easier to explain !
Thanks for your interest,
Kevin

**Comment by ceautery** [comment]
@kgf - Is this still an open issue? I ran a quick test, adding two availabilities for the same person and time, one available, the second unavailable, and the first got changed to canceled.

If this is something still going on, can you give me instructions on how to reproduce the issue?

Thanks

**Comment by kgf** [comment]
Yes, it is still open. Your test is the expected behavior. However, it doesn't consider a second model called Assignments. 

Imagine you've opened up a shelter, and you're trying to get volunteers in to staff it. You need someone at the medical intake station, and someone at the front door to greet people and direct them to the correct line. 
Now one of your volunteers sends in `Available shelter`. They'll show up as green. You assign them to medical intake. A little while later, you go to assign someone to the greeter task. This *same* person will appear as available, even though they're really not, since they're already doing something else for you.
The hardest part of this issue, I suspect, will be how to convey this info to the end user in the UI. You can't just say they're unavailable, because you might decide the greeter task is a better match to their skills, and move them over.
I hope this helps. I look forward to your ideas.

Thanks for your interest,
Kevin
